app/src/main/java/com/example/bookwormadventuresdeluxe2/RequestsFragment.java

```diff
@@ -51,23 +66,86 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         toggle.setBackground(shape);
         toggle.setOnClickListener(this);
 
-        //TODO: add booklist
-
         return view;
     }
 
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState)
+    {
+        //TODO: actually query the books
+        FirebaseFirestore rootRef = FirebaseFirestore.getInstance();
+        Query query = rootRef.collection(getString(R.string.books_collection)).orderBy("title");
+
+        FirestoreRecyclerOptions<Book> options = new FirestoreRecyclerOptions.Builder<Book>()
+                .setQuery(query, Book.class)
+                .build();
+
+        requestsRecyclerView = (RecyclerView) view.findViewById(R.id.requests_recycler_view);
+        requestsRecyclerView.setHasFixedSize(true);
+
+        requestsRecyclerLayoutManager = new LinearLayoutManager(this.getContext());
+        requestsRecyclerView.setLayoutManager(requestsRecyclerLayoutManager);
+
+        requestsRecyclerAdapter = new BookListAdapter(this.getContext(), options, R.id.requests);
+        requestsRecyclerView.setAdapter(requestsRecyclerAdapter);
+
+        Query query2 = rootRef.collection(getString(R.string.books_collection)).orderBy("title");
```

Comment:
Why are we doing 2 queries?

Reply:
there are two lists, own for requests on your books, other for your requests

Reply:
Ok these queries will need to be renamed but I think its fine for now